src/openfermion/utils/_sparse_tools.py

```diff
@@ -744,32 +744,36 @@ def expectation(operator, state):
     Raises:
         ValueError: Input state has invalid format.
     """
-
     if isinstance(state, scipy.sparse.spmatrix):
         # Handle density matrix.
         if isinstance(operator, scipy.sparse.linalg.LinearOperator):
             raise ValueError('Taking the expectation of a LinearOperator with '
                              'a density matrix is not supported.')
         product = state * operator
-        expectation = numpy.sum(product.diagonal())
+        return numpy.sum(product.diagonal())
 
-    elif isinstance(state, numpy.ndarray):
+    if isinstance(state, numpy.ndarray):
         # Handle state vector.
```

Comment:
Should remove this comment now that it includes a density matrix case.

Reply:
done